src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                    case "v141":
+                    case "v141_xp":
+                        return "19.10.00";
+                    case "v140":
+                    case "v140_xp":
+                        return "19.00.00";
+                    case "v120":
+                    case "v120_xp":
+                        return "18.00.00";
+                    case "v110":
+                    case "v110_xp":
+                        return "17.00.00";
+                    case "v100":
+                        return "16.00.00";
+                    case "v90":
+                        return "15.00.00";
+                }
+            }
+
+            internal static string ConvertCompileAsAndGetSqLanguage(string value, string path, out string sqLanguage)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported CompileAs: " + value);
+                    case "Default":
+                        // Compile files with extensions ".cpp", ".cxx" and ".cc" as Cpp and files with extension ".c" as C
+                        if (path.ToLowerInvariant().EndsWith(".cpp") || path.ToLowerInvariant().EndsWith(".cxx") || path.ToLowerInvariant().EndsWith(".cc"))
+                        {
+                            sqLanguage = CPP_LANGUAGE_KEY;
+                        }
+                        else if (path.ToLowerInvariant().EndsWith(".c"))
+                        {
+                            sqLanguage = C_LANGUAGE_KEY;
+                        }
+                        else
+                        {
+                            sqLanguage = null;
+                        }
+                        return "";
+                    case "CompileAsC":
+                        sqLanguage = "c";
+                        return "/TC";
+                    case "CompileAsCpp":
+                        sqLanguage = CPP_LANGUAGE_KEY;
+                        return "/TP";
+                }
+            }
+
+            internal static string ConvertCompileAsManaged(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported CompileAsManaged: " + value);
+                    case "":
+                    case "false":
+                        return "";
+                    case "true":
+                        return "/clr";
+                    case "Pure":
+                        return "/clr:pure";
+                    case "Safe":
+                        return "/clr:safe";
+                }
+            }
+
+            internal static string ConvertRuntimeLibrary(string value)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported RuntimeLibrary: " + value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)